services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/credentials/CredentialKey.java

```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.deviceregistry.service.credentials;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+import org.eclipse.hono.deviceregistry.service.tenant.TenantKey;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.MoreObjects.ToStringHelper;
+
+/**
+ * Provides a unique key for a <em>Credential</em> resource of Hono's
+ * <a href="https://www.eclipse.org/hono/docs/api/management/">Device Registry Management API</a>.
+ * It is used for storing and retrieving values from the backend storage and external systems.
+ */
+public final class CredentialKey implements Serializable {
+
+    private final String tenantId;
+    private final String authId;
+    private final String type;
+
+    private CredentialKey(final String tenantId, final String authId, final String type) {
+        this.tenantId = tenantId;
+        this.authId = authId;
+        this.type = type;
+    }
+
+    /**
+     * Gets the tenant identifier.
+     *
+     * @return The identifier or {@code null} if not set.
+     */
+    public String getTenantId() {
+        return tenantId;
+    }
+
+    /**
+     * Gets the authentication id of the credential.
+     *
+     * @return The authentication identifier or {@code null} if not set.
+     */
+    public String getAuthId() {
+        return authId;
+    }
+
+    /**
+     * Gets the type of the credential.
+     *
+     * @return The type or {@code null} if not set.
+     */
+    public String getType() {
+        return type;
+    }
+
+    /**
+     * Creates a credential key from tenant identifier, authentication identifier and type.
+     *
+     * @param tenantId The tenant identifier.
+     * @param authId The authentication identifier.
+     * @param type The credential type.
+     * @throws NullPointerException if any of the parameters is {@code null}.
+     * @return The credential key.
+     */
+    public static CredentialKey from(final String tenantId, final String authId, final String type) {
```

Comment:
I would recommend to name the method `credentialKey` instead of `from`. The reason for that is, when using static imports, you will get a name clash with the other `from` methods.

Reply:
Static imports are treated as errors by checkstyle plugin, so we can use them. And we have pattern of using `from()` methods in these kind of value objects throughout the codebase. Also, there was also `of()` in `TenantHandle`, so it should be consistent now.